src/requirementslib/models/url.py

```diff
@@ -42,8 +42,8 @@ def _get_parsed_url(url):
         auth, _, url = url.rpartition("@")
         url = "{scheme}://{url}".format(scheme=scheme, url=url)
         parsed = urllib3_parse(url)._replace(auth=auth)
-    if parsed.auth and unquote_plus(parsed.auth) != parsed.auth:
```

Comment:
The goal here -- and I am not sure if this is even realistic but it seemed to work ok -- was to determine if the URL had already been escaped so we didn't wind up double-escaping, i.e. by unquoting to see if that changes anything. I suppose the better check here would be to see if there are any illegal characters...

open to suggestions on this, but also open to moving forward with this change. @frostming - thoughts?

Reply:
I remove this judgment because it seems meaningless to me:

If they are unequal, we perform the replacement, but if the result is equal, we can also do it since the replacement just does nothing.

Reply:
oh... right... duh